server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -367,6 +365,8 @@ private void balanceByWeights() {
                 final BalancedShardsAllocator.ModelNode maxNode = modelNodes[highIdx];
                 advance_range: if (maxNode.numShards(index) > 0) {
                     final float delta = absDelta(weights[lowIdx], weights[highIdx]);
+                    final int finalHighIdx = highIdx;
```

Comment:
Instead of creating new final variables to be used, what about creating 2 private utilities like `log_stop_balancing/balancing(final variables...)` and then just call those utilities instead. I don't want us to create noise changes in code logic.

Reply:
If I understood you correctly, you are suggesting like having a method like below for "lowIdx"

final int getLowIndex(final int index) {
        return index;
    }

and invoking this method from lambda expression like "getLowIndex(lowIdx)"
This won't be possible, because again "lowIdx" is non-final and we can not refer that from lambda expression.
If you meant something else please correct me.

Reply:
No, abstracting out entire logging statement itself.

```
private log_stop_balancing(final int highIdx, final int lowIdx) {
     logger.trace(
                                 "Stop balancing index [{}]  min_node [{}] weight: [{}]" + "  max_node [{}] weight: [{}]  delta: [{}]",
                            () -> index,
                            () -> maxNode.getNodeId(),
                            () -> weights[finalHighIdx],
                            () -> minNode.getNodeId(),
                            () -> weights[finalLowIdx],
                            () -> delta
}

private log_balancing(final int highIdx, final int lowIdx) {
...
}
```

and then just calling this utility method.